pkg/provisioner/ironic/ironic.go

```diff
@@ -1165,10 +1165,10 @@ func (p *ironicProvisioner) GetFirmwareComponents() ([]metal3api.FirmwareCompone
 	// Get the components from Ironic via Gophercloud
 	componentList, componentListErr := nodes.ListFirmware(p.ctx, p.client, ironicNode.UUID).Extract()
 
-	if componentListErr != nil {
+	if componentListErr != nil || len(componentList) == 0 {
 		bmcAccess, _ := p.bmcAccess()
-		if bmcAccess.FirmwareInterface() == "no-firmware" {
-			return nil, fmt.Errorf("node %s is using firmware interface %s: %w", ironicNode.UUID, bmcAccess.FirmwareInterface(), componentListErr)
+		if bmcAccess.FirmwareInterface() != "redfish" {
```

Comment:
This feels very-very wrong. We don't assume redfish is and will be the only driver to support firmware updates.

Reply:
Since `bmcAccess.FirmwareInterface` is something we control, I think it's safe to assume that we use "no-firmware" for any drivers that do not support the operation.

Reply:
I changed because I noticed we are returning "" instead of "no-firmware"  for the `pkg/hardwareutils/bmc` (should I also update in this PR and go back to the old condition?

Reply:
Okay, we can use `ironicNode` then. It will have the real value.